docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,176 @@
+    Controller Manager   v1.9.0    v1.9.2
+    Scheduler            v1.9.0    v1.9.2
+    Kube Proxy           v1.9.0    v1.9.2
+    Kube DNS             1.14.5    1.14.7
+    Etcd                 3.2.7     3.1.11
+
+**Caution:** Currently the only supported configuration for kubeadm HA clusters requires the use of an externally managed etcd cluster upgrading etcd is not supported as a part of the upgrade. If necessary you will have to upgrade the etcd cluster according to [etcd's upgrade instructions](/docs/tasks/administer-cluster/configure-upgrade-etcd/), which is beyond the scope of these instructions.
+{: .caution}
+
+## Upgrading your control plane
+
+The following procedure must be applied on a single master node and repeated for each subsequent master node sequentially.
+
+Before initiating the upgrade with `kubeadm` `configmap/kubeadm-config` needs to be modified for the current master host. Replace any hard reference to a master host FQDN with the current master hosts' FQDN:
+
+```shell
+kubectl get configmap -n kube-system kubeadm-config -o yaml >/tmp/kubeadm-config-cm.yaml
+sed -i 's/^\([ \t]*nodeName:\).*/\1 <CURRENT-MASTER-FQDN>/' /tmp/kubeadm-config-cm.yaml
+kubectl apply -f /tmp/kubeadm-config-cm.yaml --force
+```
+
+Now the upgrade process can start. Use the target version determined in the preparation step and run the following command (press “y” when prompted):
+
+```shell
+/tmp/kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+If the operation was successful you’ll get a message like this:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.9.2". Enjoy!
+
+    To upgrade the cluster with CoreDNS as the default internal DNS, invoke `kubeadm upgrade apply` with the `--feature-gates=CoreDNS=true` flag.
+
+`kubeadm upgrade apply` does the following:
+
+- Checks that your cluster is in an upgradeable state:
+  - The API server is reachable,
+  - All nodes are in the `Ready` state
+  - The control plane is healthy
+- Enforces the version skew policies.
+- Makes sure the control plane images are available or available to pull to the machine.
+- Upgrades the control plane components or rollbacks if any of them fails to come up.
+- Applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
+- Creates new certificate and key files of apiserver and backs up old files if they're about to expire in 180 days.
+
+Next, manually upgrade your CNI provider
+
+Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow. Check the [addons](/docs/concepts/cluster-administration/addons/) page to find your CNI provider and see if there are additional upgrade steps necessary.
+
+**Note:** The `kubeadm upgrade apply` step has been known to fail when run initially on the secondary masters (timed out waiting for the restarted static pods to come up). It should succeed if retried after a minute or two.
```

Comment:
I haven't actually seen this yet, but it makes sense that it happens. Is there more specific instructions we can provide to users, such as "wait for X to become ready before moving on to the secondary masters"?

Reply:
Nothing that I am aware of.